relay/prover.go

```diff
@@ -379,6 +449,61 @@ func (pr *Prover) ProveHostConsensusState(ctx core.QueryContext, height exported
 	return pr.originProver.ProveHostConsensusState(ctx, height, consensusState)
 }
 
+// shouldUseSHFUGRPC determines whether to use SHFU gRPC server based on config and environment variable
+// Environment variable YRLY_LCP_SHFU_GRPC_ENABLE=yes enables gRPC, disabled by default
+func (pr *Prover) shouldUseSHFUGRPC() (bool, string) {
+	// First check if address is configured
+	if pr.config.ShfuGrpcAddress == "" {
+		// No address configured, cannot use gRPC
+		return false, ""
+	}
+
+	// Check if gRPC is enabled via environment variable
+	envEnable := os.Getenv("YRLY_LCP_SHFU_GRPC_ENABLE")
+	if envEnable != "yes" {
+		// Environment variable is not "yes", gRPC disabled by default
+		return false, ""
+	}
+
+	// Address configured and gRPC enabled, use gRPC
+	return true, pr.config.ShfuGrpcAddress
+}
+
+// getUpdateClientFromGRPC retrieves SHFU results from gRPC server using height range
+func getUpdateClientFromGRPC(ctx context.Context, logger *log.RelayLogger, grpcAddress string, targetChain core.Chain, counterparty core.Chain, latestFinalizedHeader core.Header) ([]*shfu_storage.UpdateClientResult, error) {
```

Comment:
I think `getUpdateClientResultsFromGRPC` is more appropriate:

```suggestion
func getUpdateClientResultsFromGRPC(ctx context.Context, logger *log.RelayLogger, grpcAddress string, targetChain core.Chain, counterparty core.Chain, latestFinalizedHeader core.Header) ([]*shfu_storage.UpdateClientResult, error) {
```

Reply:
renamed